types/alpinejs/index.d.ts

```diff
@@ -269,7 +269,8 @@ export interface Alpine {
      * @param name the id of the x-data context
      * @param callback the initializer of the x-data context
      */
-    data(name: string, callback: (...initialStateArgs: unknown[]) => AlpineComponent): void;
+    // eslint-disable-next-line no-unnecessary-generics
+    data<TInitialStateArgs extends unknown[]>(name: string, callback: (...initialStateArgs: TInitialStateArgs) => AlpineComponent): void;
```

Comment:
Oh, I did not know eslint rules could be used to work around these limitations to improve developer experience!

Would you mind also adding generics to `store` (as requested in [this thread](https://github.com/alpinejs/alpine/issues/2199#issuecomment-1013682338))?